src/db/collections/outputs/mod.rs

```diff
@@ -502,178 +500,119 @@ mod analytics {
-            if index == 0 {
-                return Ok(Default::default());
-            }
-            let (start_state, end_state) =
-                tokio::try_join!(self.get_unique_nft_ids(index - 1), self.get_unique_nft_ids(index))?;
-
-            Ok(NftActivityAnalytics {
-                created_count: end_state.difference(&start_state).copied().count() as _,
-                transferred_count: start_state.intersection(&end_state).copied().count() as _,
-                destroyed_count: start_state.difference(&end_state).copied().count() as _,
-            })
-        }
-
-        pub(crate) async fn get_unique_foundry_ids(&self, index: MilestoneIndex) -> Result<HashSet<FoundryId>, Error> {
-            #[derive(Deserialize)]
-            struct FoundryIdsResult {
-                id: FoundryId,
-            }
-
-            self.aggregate::<FoundryIdsResult>(
-                vec![
-                    doc! { "$match": {
-                        "output.kind": "foundry",
-                        "metadata.booked.milestone_index": { "$lte": index },
-                        "metadata.spent_metadata.spent.milestone_index": { "$not": { "$lte": index } }
-                    } },
-                    doc! { "$project": {
-                        "id": "$output.foundry_id"
-                    } },
-                ],
-                None,
-            )
-            .await?
-            .map_ok(|res| res.id)
-            .try_collect()
-            .await
-        }
-
-        /// Gathers unique foundry ids that were created/transferred/burned in the given milestone.
-        #[tracing::instrument(skip(self), err, level = "trace")]
-        pub async fn get_foundry_output_analytics(
-            &self,
-            index: MilestoneIndex,
-        ) -> Result<FoundryActivityAnalytics, Error> {
-            if index == 0 {
-                return Ok(Default::default());
-            }
-            let (start_state, end_state) = tokio::try_join!(
-                self.get_unique_foundry_ids(index - 1),
-                self.get_unique_foundry_ids(index)
-            )?;
-            Ok(FoundryActivityAnalytics {
-                created_count: end_state.difference(&start_state).copied().count() as _,
-                transferred_count: start_state.intersection(&end_state).copied().count() as _,
-                destroyed_count: start_state.difference(&end_state).copied().count() as _,
-            })
+            Ok(self
+                .aggregate(
+                    vec![
+                        doc! { "$match": {
+                            "output.kind": "nft",
+                            "$or": [
+                                { "metadata.booked.milestone_index": index },
+                                { "metadata.spent_metadata.spent.milestone_index": index },
+                            ],
+                        } },
+                        doc! { "$facet": {
```

Comment:
no way around `$facet`? (I remember you found a solution to circumvent it last time)

Reply:
The solution is writing multiple queries. However, in this case the index being used is before the facet, so performance-wise it should not be an issue.